py-rse/bash-basics.Rmd

```diff
@@ -1292,33 +1329,33 @@ it reads the output of the upstream command.
 
 We can use `|` to build pipes of any length.
 For example,
-we can use the command `head` to get just the first three lines of sorted data:
+we can use the command `head` to get just the first three lines of sorted data, the three shortest books:
 
 ```shell
-$ wc -l *.csv | sort -n | head -n 3
+$ wc -l *.txt | sort -n | head -n 3
 ```
 
 ```text
-    8314 bellambi_temp.csv
-    8315 bellambi_prec.csv
-    8680 tuggeranong_temp.csv
+    2660 common_sense.txt
+    3582 time_machine.txt
+    4102 life_of_frederick_douglass.txt
 ```
 
 If we want the last line of this,
 we can add a call to the `tail` command:
 
 ```shell
-$ wc -l *.csv | sort -n | head -n 3 | tail -n 1
+$ wc -l *.txt | sort -n | head -n 3 | tail -n 1
 ```
 
 ```text
-    8680 tuggeranong_temp.csv
+  4102 life_of_frederick_douglass.txt
```

Comment:
It's a little weird that the final result we save is the name of the third shortest book, because in the text earlier on we ask the question: "which of these books is shortest?"

Perhaps it's better not to introduce the final call to `tail` and instead have the final thing be:
`$ wc -l *.txt | sort -n | head -n 1 > shortest.txt` 
(i.e. the shortest book)

Reply:
true, although I thought the purpose is to show the pipe can be any length (multiple commands), but I guess they get the message with two commands as well, you're right, it's better than suddenly selecting the third shortest book!